nmpolicy/internal/ast/types.go

```diff
@@ -0,0 +1,16 @@
+package ast
+
+type Meta struct {
+	Position int `yaml:"pos"`
+}
+
+type TernaryOperator [3]Node
+type VariadicOperator []Node
+
+type Node struct {
+	Meta     `yaml:",inline"`
+	EqFilter *TernaryOperator  `yaml:"eqfilter,omitempty"`
+	Path     *VariadicOperator `yaml:"path,omitempty"`
+	String   *string           `yaml:"string,omitempty"`
```

Comment:
Can this be placed under a `Value` struct where it will consist of all the other types? Then just embed it here.
If having a struct does not sound right to you, perhaps just place this separated from the rest so we will understand it is "something else".

Reply:
I have try grouping them on `Terminal` struct and embedding the struct, it complicate a little the use since golang force yo to use to initialize the ast.Terminal.

Before
```golang
{Identity: strPtr("currentState")}
```
If we move Identity and String into a Terminal and embeed terminal we are forced to do 
```golang
{Terminal: ast.Terminal{Identity: strPtr("currentState")}}
```
So I don't know if it worth it.